SEIR/NPI/Stacked.py

```diff
@@ -1,58 +1,67 @@
-import functools
+import warnings
+
+import confuse
 import pandas as pd
 
 from .base import NPIBase
 
 REDUCE_PARAMS = ["alpha", "r0", "gamma", "sigma"]
 
+
 class Stacked(NPIBase):
-    def __init__(self, *, npi_config, global_config, geoids, loaded_df = None):
-        super().__init__(npi_config)
+    def __init__(self, *, npi_config, global_config, geoids, loaded_df=None):
+        super().__init__(name=npi_config.name)
 
         self.start_date = global_config["start_date"].as_date()
         self.end_date = global_config["end_date"].as_date()
 
         self.geoids = geoids
+        self.reductions = {param: None for param in REDUCE_PARAMS}  # lazily construct: None means no reduction
+        self.reduction_params = []
 
-        # Gather parameter reductions
-        reduction_lists = {}
-        for param in REDUCE_PARAMS:
-            reduction_lists[param] = []
-
-        self.sub_npis = []
```

Comment:
We use this for inference.

Reply:
If we're going this route, can we just make get() lazy instead?

Reply:
what do you use for inference? i can't find reference in other parts of the repo

Reply:
Think i just missed that you changed reduction_lists to reductions.  Anyway, the only place this is important is in the part you and @kkintaro are already talking about, so we can continue there